contrib/pipelines/secadvisor/README.md

```diff
@@ -2,33 +2,223 @@
+## Setup Object Store
 
-## Setup Pipeline
+The security advisor saves data to an S3-type object store.
+The parameters to access the object store must be provided in the secadvisor.yml configuration file.
+In case the user does not already have an object store, we show below how to create an object store for testing purposes.
+We show how to create either a minio object store running locally or an IBM Cloud Object Store.
 
-Build and run the pipeline:
+You can either use AWS-style HMAC keys by setting "access_key" and "secret_key", or use IBM IAM OAuth by setting "api_key".
+
+### Setup IBM COS
+
+In IBM Cloud, Create an Object Store resource. The Lite plan gives limited resources for free.
+In the Object Store, create a bucket to hold the Skydive flow information. For simple testing, you can choose Single Site resiliency.
+Look under Bucket Configuration to see the Public endpoint (url) where the bucket is accessed. This endpoint information needs to go into the secadvisor.yml "endpoint" field.
+Go to the "Service Credentials" panel and create a new credential. Click on "View Credential" to get the details of the credential. The apikey needs to go into the secadvisor.yml "api_key" field.
+In the secadvisor.yml file, uncomment the iam_endpoint field and set it to https://iam.cloud.ibm.com/identity/token.
+
+### Setup Minio
+
+For running tests on a local machine, it is possible to set up a local Minio object store.
+
+Follow the instructions on https://docs.min.io/docs/minio-quickstart-guide.html to install and run a minio object store (an AWS S3-like object store).
+
+```
+wget https://dl.min.io/server/minio/release/linux-amd64/minio
+chmod +x minio
+./minio server /data
+```
+
+This will run a basic minio object store server on your machine, placing the objects in "/data".
+Note the printed output in the logs from the ./minio command and take down the values of "Endpoint", "AccessKey", and "SecretKey".
+Place these items in the secadvisor.yml in the appropriate locations.
+Create a bucket in the object store to hold the security-advisor data.
+
+## Run the security advisor pipeline
+
+Build and run the pipeline in the secadvisor directory:
 
 ```
+cd $SKYDIVE_BASE/contrib/pipelines/secadvisor
 make all
 make install
 make run
 ```
 
+or
+
+```
+make
+secadvisor secadvisor.yml
+```
+
+The "make all" compiles the source code and creates the secadvisor binary file in the same directory.
+The "make run" runs the secadvisor binary using the secadvisor.yml.default that is provided with the source code.
+This will run the secadvisor pipeline, which will ultimately fail to write to the object store,
+unless the user first defines valid object store parameters in the secadvisor.yml file.
+
 ## Generate Flows
 
 Via the Skydive WebUI setup captures and generate traffic which should
 result in the secadvisor pipeline sending flows to the ObjectStore.
 
+For example, run some iperf traffic between entities and capture the flow via the Skydive GUI.
```

Comment:
please generate traffic via traffic injection feature of Skydive, a better alternative to the WebUI is using REST commands (or `skydive client`) for setup.